Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs

```diff
@@ -11,53 +10,173 @@
 
 ////TODO: property that tells whether a Touchscreen supports pressure
 
-////TODO: if activeTouches is called multiple times in a single frame, only update the array once
+////TODO: add support for screen orientation
 
-////TODO: click detection / primaryaction handling
+////TODO: touch is hardwired to certain memory layouts ATM; either allow flexibility or make sure the layouts cannot be changed
 
-////TODO: add orientation
+////TODO: startTimes are baked *external* times; reset touch when coming out of play mode
 
-////TODO: look at how you can meaningfully use touches with actions
+////REVIEW: where should we put handset vibration support? should that sit on the touchscreen class? be its own separate device?
 
-////TODO: touch is hardwired to certain memory layouts ATM; either allow flexbility or make sure the layouts cannot be changed
-
-//// Remaining things to sort out around touch:
-//// - How do we handle mouse simulation?
-//// - How do we implement click-detection for touch?
-//// - High frequency touches
-//// - Touch prediction
+// What I want:
+// X There should always be a reliable "primary" touch
+// X Touch #0..#n should always correspond to first..last touch
+// - Reliable binding from actions
+// X TouchManager should make tracking touches super easy
+// - Lots of tests to cover all this
+// - Ideally, make touch not depend on hardcoded state format
 
 namespace UnityEngine.InputSystem.LowLevel
 {
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1726:UsePreferredTerms", MessageId = "Flags", Justification = "Fix this after landing Touch refactor")]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1717:OnlyFlagsEnumsShouldHavePluralNames", Justification = "Fix this after landing Touch refactor")]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1028:EnumStorageShouldBeInt32", Justification = "byte to correspond to TouchState layout.")]
+    [Flags]
+    public enum TouchFlags : byte
```

Comment:
Maybe just make this internal? Then the name does not matter - users have nicer properties to access the individual bits anyways, and should not bother with how it's stored.

Reply:
Yup makes sense. Done.